python/cudf/cudf/core/column/column.py

```diff
@@ -1041,101 +852,67 @@ def column_empty_like_same_mask(column, dtype):
         The dtype of the data buffer.
     """
     result = column_empty_like(column, dtype)
-    if column.has_null_mask:
-        result = result.set_mask(column.mask)
+    if column.mask:
+        result.mask = column.mask
     return result
 
 
-def column_empty(row_count, dtype, masked, categories=None):
+def column_empty(row_count, dtype, masked, categories=None, name=None):
     """Allocate a new column like the given row_count and dtype.
     """
     dtype = pd.api.types.pandas_dtype(dtype)
-    if masked:
-        mask = cudautils.make_empty_mask(row_count)
-    else:
-        mask = None
-
-    if categories is None and is_categorical_dtype(dtype):
-        categories = [] if dtype.categories is None else dtype.categories
+    offsets = None
 
-    if categories is not None:
-        mem = rmm.device_array((row_count,), dtype=dtype)
-        data = Buffer(mem)
-        dtype = "category"
+    if is_categorical_dtype(dtype):
+        data = Buffer.empty(row_count * dtype.data_dtype.itemsize)
     elif dtype.kind in "OU":
-        if row_count == 0:
-            data = nvstrings.to_device([])
-        else:
-            mem = rmm.device_array((row_count,), dtype="float64")
-            data = nvstrings.dtos(mem, len(mem), nulls=mask, bdevmem=True)
+        data = Buffer.empty(row_count)
+        offsets = Buffer.empty((row_count + 1) * np.dtype("int32").itemsize)
     else:
-        mem = rmm.device_array((row_count,), dtype=dtype)
-        data = Buffer(mem)
+        data = Buffer.empty(row_count * dtype.itemsize)
 
-    if mask is not None:
-        mask = Buffer(mask)
+    if masked:
+        mask = Buffer.from_array_like(cudautils.make_empty_mask(row_count))
+    else:
+        mask = None
 
-    from cudf.core.column import build_column
+    return build_column(data, dtype, mask, offsets=offsets, name=name)
 
-    return build_column(data, dtype, mask, categories)
 
+def build_column(data, dtype, mask=None, offsets=None, name=None):
```

Comment:
Discussed offline with Ashwin and we're going to update this signature to:
```
def _build_column(data, dtype, mask=None, children=[], offset=0, name=None):
```

Reply:
Done